lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -154,35 +162,60 @@ defaultRetryPolicy =
 
 -- | Subscribe to a blockchain and get called with new block (in order)!
 follow
-    :: NetworkLayer IO tx block
+    :: Show e
+    => NetworkLayer IO tx block
     -- ^ The @NetworkLayer@ used to poll for new blocks.
     -> Trace IO Text
     -> BlockHeader
     -- ^ The local tip to start at. Blocks /after/ the tip will be yielded.
     -> (NE.NonEmpty block -> BlockHeader -> ExceptT e IO ())
-     -- ^ Callback with blocks and the current tip of the /node/. @follow@ stops
-     -- polling and terminates if the callback errors.
+    -- ^ Callback with blocks and the current tip of the /node/. @follow@ stops
+    -- polling and terminates if the callback errors.
     -> (block -> BlockHeader)
+    -- ^ Getter on the abstract 'block' type
     -> IO ()
-follow nl tr start yield header = do
-    liftIO $ runExceptT (networkTip nl) >>= \case
-        Right nwTip ->
-            restoreStep (start, nwTip)
-        Left e -> do
-            logError tr $ "Failed to get network tip: " +|| e ||+ ""
-            restoreSleep start
+follow nl tr start yield header =
+    sleep 0 start
```

Comment:
I've made the delay an argument of sleep and removed some logic duplication here.